### PR TITLE
Fixed Phalcon\Config\Adapter\Ini() to handle arrays correctly

### PR DESCRIPTION
**Pull request by zsilbi**
Hello!

* Type: bug fix
* Link to issue: https://github.com/phalcon/cphalcon/issues/14025

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I updated the CHANGELOG

Small description of change:

Added a return in cast() for arrays.

Thanks,
zsilbi

### CODE REVIEW

**Review by sergeyklay — DISMISSED** [review]

**Comment by sergeyklay** [comment]
Could you please rebase

File: tests/_data/assets/config/config-with-constants.ini [hunk 1]
```diff
@@ -6,3 +6,5 @@ test = TEST_CONST
 path = TEST_CONST"/another-thing/somewhere"
 parent.property = TEST_CONST
 parent.property2 = TEST_CONST"hello"
+testArray[] = value1
+testArray[] = value2
```
Comment (Jeckerson) [inline tests/_data/assets/config/config-with-constants.ini]:
```suggestion
testArray[] = value2

```

**Review by Jeckerson — DISMISSED** [review]

**Comment by zsilbi** [comment]
@sergeyklay rebased, thanks

**Review by Jeckerson — APPROVED** [review]

**Review by niden — APPROVED** [review]

**Comment by niden** [comment]
Thank you @zsilbi

**Pull request merged**